pilot/pkg/model/push_context.go

```diff
@@ -768,6 +771,52 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	return gwSvcs
 }
 
+// add services from MeshConfig.ExtensionProviders
+// TODO: include cluster from EnvoyFilter such as global ratelimit [demo](https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#global-rate-limit)
+func (ps *PushContext) getHostsFromMeshConfig(proxy *Proxy) sets.Set {
+	hostsFromMeshConfig, applicableProviders := sets.NewSet(), sets.NewSet()
+
+	for _, p := range ps.Telemetry.AccessLogging(proxy).Providers {
+		if p.Name != "" {
+			applicableProviders.Insert(p.Name)
+		}
+	}
+
+	if tracingCfg := ps.Telemetry.Tracing(proxy); tracingCfg != nil {
+		applicableProviders.Insert(tracingCfg.Provider.Name)
+	}
+
+	for _, prov := range ps.Mesh.ExtensionProviders {
+		if !applicableProviders.Contains(prov.Name) {
+			continue
+		}
+
+		switch p := prov.Provider.(type) {
```

Comment:
How do we keep this list in sync with API? any possibility?

Reply:
maybe `refelect`? assume there's alway a field named `Address`.

Reply:
I think we can do

```
type Servicer interface {
  GetService() string
}
p, ok := prov.Provider.(Servicer)
if ok {
  insert(p.GetService())
}
```

Not 100% sure though.

If not, I would add a test that uses reflect and compares it here (rather than reflect in the prod code)

Reply:
1. first of all:
```
	for _, prov := range ps.Mesh.ExtensionProviders {
		if p, ok := prov.Provider.(ExtensionProvider); ok {
			hostsFromMeshConfig.Insert(p.GetService())
		}
	}
```
the code will not working.

```goalng
case *meshconfig.MeshConfig_ExtensionProvider_EnvoyExtAuthzHttp:
	hostsFromMeshConfig.Insert(p.EnvoyExtAuthzHttp.Service)
case *meshconfig.MeshConfig_ExtensionProvider_EnvoyExtAuthzGrpc:
	hostsFromMeshConfig.Insert(p.EnvoyExtAuthzGrpc.Service)
```

`GetService()`  is not function of `Provider`.

2. `I would add a test that uses reflect and compares it here (rather than reflect in the prod code)` this may block `Automator: update istio/api@master` work.